### PR TITLE
ICU-13833 Fixing doxygen warnings in numberformatter.h and other files.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed at https://unicode-org.atlassian.net :  ICU-13833
- [x] Update PR title to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/i18n/unicode/numberformatter.h [hunk 1]
```diff
@@ -146,12 +146,29 @@ class GeneratorHelpers;
...
+ */
 void touchRangeLocales(impl::RangeMacroProps& macros);
 
 } // namespace impl
 
-// Reserve extra names in case they are added as classes in the future:
+/**
+ * Extra name reserved in case it is needed in the future.
+ *
+ * @draft ICU 63
```
Comment (aheninger) [inline icu4c/source/i18n/unicode/numberformatter.h]: I don't see an API proposal for this.
Should it be \@internal for now?

File: icu4c/source/i18n/unicode/numberformatter.h [hunk 2]
```diff
@@ -146,12 +146,29 @@ class GeneratorHelpers;
...
+ */
 typedef Notation CompactNotation;
+
+/**
+ * Extra name reserved in case it is needed in the future.
+ *
+ * @draft ICU 63
```
Comment (aheninger) [inline icu4c/source/i18n/unicode/numberformatter.h]: same as above

File: icu4c/source/i18n/unicode/numberformatter.h (hunk 1 above)
Comment (sffc) [reply icu4c/source/i18n/unicode/numberformatter.h]: The type name is returned from @draft methods.

File: icu4c/source/i18n/unicode/numberformatter.h (hunk 2 above)
Comment (sffc) [reply icu4c/source/i18n/unicode/numberformatter.h]: As above, the type name is returned from @draft methods, so it does not make sense for it to be anything but @draft.

**Pull request merged**